server/signal.go

```diff
@@ -11,12 +11,12 @@ import (
 
 // Signal Handling
 func (s *Server) handleSignals() {
-	if s.opts.NoSigs {
+	if s.getOpts().NoSigs {
 		return
 	}
 	c := make(chan os.Signal, 1)
 
-	signal.Notify(c, syscall.SIGINT, syscall.SIGUSR1)
+	signal.Notify(c, syscall.SIGINT, syscall.SIGUSR1, syscall.SIGHUP)
```

Comment:
I agree with @derekcollison. let's build up the features for supporting it and then add in the reload once we have the critical items (Certs and Auth).

Derek, you agree we can merge all the bits to support in prior to having the signal and making the feature available in the server, correct?